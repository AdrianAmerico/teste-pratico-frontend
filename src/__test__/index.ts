export * from './global-setup';
export * from './sut-mock-provider';
export * from './factory';