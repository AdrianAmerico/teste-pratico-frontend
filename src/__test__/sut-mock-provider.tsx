import { ReactNode } from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

export const sutMockProvider = (children: ReactNode) => {
  return render(
    <MemoryRouter>
      <QueryClientProvider client={new QueryClient()}>
        {children}
      </QueryClientProvider>
    </MemoryRouter>
  );
};
