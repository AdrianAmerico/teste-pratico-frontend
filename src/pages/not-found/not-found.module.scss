@use '@/styles' as *;

.container {
  min-height: 100vh;
  background-color: $gray-00;
  display: flex;
  flex-direction: column;
}

.main {
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 4rem 1.5rem;
  text-align: center;
}

.contentWrapper {
  max-width: 28rem;
  margin: 0 auto;
}

.statusCode {
  font-size: 6rem;
  font-weight: bold;
  color: $blue-primary;
  margin-bottom: 1rem;
}

.title {
  font-size: 1.875rem;
  font-weight: bold;
  color: $black-neutral;
  margin-bottom: 1rem;
}

.description {
  color: $black-neutral;
  margin-bottom: 2rem;
  font-size: 1.125rem;
}

.actions {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  justify-content: center;

  @media (min-width: 640px) {
    flex-direction: row;
  }
}

.backButton {
  transition: ease all 200ms;
  border: 1px solid $gray-05;
  color: $black-neutral;
  background-color: $white;
  padding: 0.75rem 1.5rem;

  &:hover {
    background-color: $gray-00;
  }
}
