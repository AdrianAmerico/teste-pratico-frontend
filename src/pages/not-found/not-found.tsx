import { Link } from 'react-router-dom';
import styles from './not-found.module.scss';
import { Header } from '@/components';

export const NotFoundPage = () => {
  return (
    <div className={styles.container}>
      <Header />

      {/* Main Content */}
      <main className={styles.main}>
        <div className={styles.contentWrapper}>
          <div className={styles.statusCode}>404</div>

          <h2 className={styles.title}>Página não encontrada</h2>

          <p className={styles.description}>
            Desculpe, a página que você está procurando não existe ou foi
            movida.
          </p>

          <div className={styles.actions}>
            <Link to="/" className={styles.backButton}>
              Ir para início
            </Link>
          </div>
        </div>
      </main>
    </div>
  );
};
