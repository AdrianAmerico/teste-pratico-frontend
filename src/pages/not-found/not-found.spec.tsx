import { sutMockProvider } from '@/__test__';
import { cleanup } from '@testing-library/react';
import { NotFoundPage } from './not-found';

describe('NotFoundPage', () => {
  afterEach(cleanup);

  const makeSut = () => {
    const sut = sutMockProvider(<NotFoundPage />);

    return {
      sut,
    };
  };

  it('should render correctly', () => {
    const { sut } = makeSut();

    expect(sut.container).toBeInTheDocument();
  });

  it('shoudl match snapshot', () => {
    const { sut } = makeSut();

    expect(sut.container).toMatchSnapshot();
  });
});
