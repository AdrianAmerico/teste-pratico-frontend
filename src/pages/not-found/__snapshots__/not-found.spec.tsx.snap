// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundPage > shoudl match snapshot 1`] = `
<div>
  <div
    class="_container_84634a"
  >
    <div
      class="_header_57bb28"
    >
      <a
        class="_logoLink_57bb28"
        data-discover="true"
        href="/"
      >
        <img
          alt="BeTalent Logo"
          class="_logo_57bb28"
          src="betech-logo.png"
        />
      </a>
    </div>
    <main
      class="_main_84634a"
    >
      <div
        class="_contentWrapper_84634a"
      >
        <div
          class="_statusCode_84634a"
        >
          404
        </div>
        <h2
          class="_title_84634a"
        >
          Página não encontrada
        </h2>
        <p
          class="_description_84634a"
        >
          Desculpe, a página que você está procurando não existe ou foi movida.
        </p>
        <div
          class="_actions_84634a"
        >
          <a
            class="_backButton_84634a"
            data-discover="true"
            href="/"
          >
            Ir para início
          </a>
        </div>
      </div>
    </main>
  </div>
</div>
`;
